src/lib/seam/components/DeviceDetails/LockDeviceDetails.tsx

```diff
@@ -0,0 +1,150 @@
+import classNames from 'classnames'
+import type { LockDevice } from 'seamapi'
+
+import { ChevronRightIcon } from 'lib/icons/ChevronRight.js'
+import { useAccessCodes } from 'lib/seam/access-codes/use-access-codes.js'
+import { AccessCodeTable } from 'lib/seam/components/AccessCodeTable/AccessCodeTable.js'
+import { DeviceModel } from 'lib/seam/components/DeviceDetails/DeviceModel.js'
+import { useToggleLock } from 'lib/seam/devices/use-toggle-lock.js'
+import { Alerts } from 'lib/ui/Alert/Alerts.js'
+import { Button } from 'lib/ui/Button.js'
+import { BatteryStatus } from 'lib/ui/device/BatteryStatus.js'
+import { DeviceImage } from 'lib/ui/device/DeviceImage.js'
+import { OnlineStatus } from 'lib/ui/device/OnlineStatus.js'
+import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
+import { useToggle } from 'lib/ui/use-toggle.js'
+export function LockDeviceDetails(props: {
+  device: LockDevice
+  onBack?: () => void
+  className?: string
+}): JSX.Element | null {
+  const { device, onBack, className } = props
+
+  const [accessCodesOpen, toggleAccessCodesOpen] = useToggle()
+  const toggleLock = useToggleLock(device)
+  const { accessCodes } = useAccessCodes({
+    device_id: device.device_id,
+  })
+
+  const lockStatus = device.properties.locked ? t.locked : t.unlocked
+  const toggleLockLabel = device.properties.locked ? t.unlock : t.lock
+
+  const accessCodeCount = accessCodes?.length
+
+  const accessCodeLength = device.properties?.supported_code_lengths?.[0]
```

Comment:
I think this is better logic than 

```
const accessCodeLength = device.properties?.schlage_metadata?.access_code_length
```

Reply:
yes, should always use the top-level property if available

I think `device.properties?.schlage_metadata?.access_code_length` is an artifact from before we had a standard top-level property

Reply:
Actually, I thought a bit more and we can get min / max do to `a range :)